test/integration/builder/unions.js

```diff
@@ -4,7 +4,7 @@
 
 module.exports = function(knex) {
   describe('unions', function() {
-    it('handles unions', function() {
+    it('handles unions with a callback', function() {
```

Comment:
Since neither of these tests make assertions on the result I think either adding assertions of the resultset or adding corresponding unit tests to ensure generated SQL is what's to be expected.

I see the latter has been done for some of the new tests, but not all. For example raw and pure functions unit tests are missing.

Reply:
I assumed since none of these tests have assertions on the resultset that we were just detecting if it threw.

I'm very happy to add the other unit test, though. I'll go ahead and cover each possible case!

Reply:
⚡️ cccae9a

Instead of writing a ton of repetitive tests, I opted to just mix the types for the array and multi-argument cases. Let me know if that feels comprehensive enough or if we ought to add a whole section testing each of the possible cases. I think there are 9 total: single callback, builder, raw; array callback, builder, raw; multi-argument callback, builder, raw.

Reply:
Looks good to me 👍